core/src/test/scala/unit/kafka/admin/AclCommandTest.scala

```diff
@@ -82,45 +83,73 @@ class AclCommandTest extends ZooKeeperTestHarness with Logging {
       ProducerResourceToAcls(enableIdempotence = true).getOrElse(k, Set.empty[Acl])) }
   )
 
-  @Test
-  def testAclCli() {
-    val brokerProps = TestUtils.createBrokerConfig(0, zkConnect)
+  private var brokerProps: Properties = _
+  private var zkArgs: Array[String] = _
+
+  @Before
+  override def setUp(): Unit = {
+    super.setUp()
+
+    brokerProps = TestUtils.createBrokerConfig(0, zkConnect)
     brokerProps.put(KafkaConfig.AuthorizerClassNameProp, "kafka.security.auth.SimpleAclAuthorizer")
-    val args = Array("--authorizer-properties", "zookeeper.connect=" + zkConnect)
 
+    zkArgs = Array("--authorizer-properties", "zookeeper.connect=" + zkConnect)
+  }
+
+  @Test
+  def testAclCli() {
     for ((resources, resourceCmd) <- ResourceToCommand) {
       for (permissionType <- PermissionType.values) {
         val operationToCmd = ResourceToOperations(resources)
         val (acls, cmd) = getAclToCommand(permissionType, operationToCmd._1)
-          AclCommand.main(args ++ cmd ++ resourceCmd ++ operationToCmd._2 :+ "--add")
+          AclCommand.main(zkArgs ++ cmd ++ resourceCmd ++ operationToCmd._2 :+ "--add")
           for (resource <- resources) {
-            withAuthorizer(brokerProps) { authorizer =>
+            withAuthorizer() { authorizer =>
               TestUtils.waitAndVerifyAcls(acls, authorizer, resource)
             }
           }
 
-          testRemove(resources, resourceCmd, args, brokerProps)
+          testRemove(resources, resourceCmd, brokerProps)
       }
     }
   }
 
   @Test
   def testProducerConsumerCli() {
-    val brokerProps = TestUtils.createBrokerConfig(0, zkConnect)
-    brokerProps.put(KafkaConfig.AuthorizerClassNameProp, "kafka.security.auth.SimpleAclAuthorizer")
-    val args = Array("--authorizer-properties", "zookeeper.connect=" + zkConnect)
-
     for ((cmd, resourcesToAcls) <- CmdToResourcesToAcl) {
       val resourceCommand: Array[String] = resourcesToAcls.keys.map(ResourceToCommand).foldLeft(Array[String]())(_ ++ _)
-      AclCommand.main(args ++ getCmd(Allow) ++ resourceCommand ++ cmd :+ "--add")
+      AclCommand.main(zkArgs ++ getCmd(Allow) ++ resourceCommand ++ cmd :+ "--add")
       for ((resources, acls) <- resourcesToAcls) {
         for (resource <- resources) {
-          withAuthorizer(brokerProps) { authorizer =>
+          withAuthorizer() { authorizer =>
             TestUtils.waitAndVerifyAcls(acls, authorizer, resource)
           }
         }
       }
-      testRemove(resourcesToAcls.keys.flatten.toSet, resourceCommand ++ cmd, args, brokerProps)
+      testRemove(resourcesToAcls.keys.flatten.toSet, resourceCommand ++ cmd, brokerProps)
+    }
+  }
+
+  @Test
+  def testAclsOnPrefixedResources(): Unit = {
+    val cmd = Array("--allow-principal", principal.toString, "--producer", "--topic", "Test-", "--resource-name-type", "Prefixed")
```

Comment:
Test- => Test-topic ?

Reply:
Again, this is intentional as its adding a ACLs on a prefixed resource name.